pyiceberg/table/inspect.py

```diff
@@ -405,13 +419,19 @@ def _partition_summaries_to_rows(
                     "partition_summaries": _partition_summaries_to_rows(specs[manifest.partition_spec_id], manifest.partitions)
                     if manifest.partitions
                     else [],
-                })
+                }
+                if is_all_manifests_table:
+                    manifest_row["reference_snapshot_id"] = snapshot.snapshot_id
+                manifests.append(manifest_row)
 
         return pa.Table.from_pylist(
             manifests,
-            schema=manifest_schema,
+            schema=get_all_manifests_schema() if is_all_manifests_table else get_manifests_schema(),
         )
 
+    def manifests(self) -> "pa.Table":
```

Comment:
wdyt about adding an optional `snapshot_id` here? To allow users to look at the manifest for a specific snapshot, with the added benefit to iterate over all snapshot ids for `all_manifests`

Reply:
Yeah, I am aligned with this.
But there are two parameters that I'm passing to `_generate_manifests_table` method - snapshot_id and a boolean flag whether the output is for all_manifests table which add the additional column to `all_manifests` table.
So I'll need to add this second parameter for manifests method as well. Thoughts?